File: week14/js/MajorityElement.java
	public int majorityElement(int[] nums) {
		int majorityCount = nums.length / 2;
		
        Map<Integer, Integer> counter = new HashMap<>();
		
        for (int i = 0; i < nums.length; i++) {
			counter.compute(nums[i], (key, old) -> old == null ? 1 : old + 1);
			if (counter.get(nums[i]) > majorityCount) {
				return nums[i];
			}
		}

		return -1;
	}

File: S02/week14/mk/majority_element.go
// 15ms, 6.26mb
package main

import (
	"fmt"
	"sort"
)

func main() {
	fmt.Println(majorityElement([]int{2, 2, 1, 1, 1, 2, 2}))
}

func majorityElement(nums []int) int {
	sort.Ints(nums)
	return nums[len(nums)/2]
}
